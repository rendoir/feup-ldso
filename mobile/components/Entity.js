import React from 'react';
import {
    StyleSheet
} from 'react-native';

import { Text, ListItem, Button } from 'native-base';

export default class Entity extends React.Component {

    constructor(props) {
        super(props);

    }

    render() {
        return (
            <ListItem style={styles.listItem} onPress={this.props.onPress} >
                <Button transparent onPress={this.props.onPress}><Text style={styles.buttonText}>{this.props.data.initials}</Text></Button>
            </ListItem>
        );
    }
}

const styles = StyleSheet.create({
    buttonText: {
        color: 'white',
        fontFamily: 'OpenSans-Regular',
        fontSize: 18,
        paddingLeft: 5
    },
    listItem: {
        justifyContent: "center",
        backgroundColor: '#7C8589',
        width: '91%',
        borderBottomWidth: 1,
        borderBottomColor: 'white'
    }
});
