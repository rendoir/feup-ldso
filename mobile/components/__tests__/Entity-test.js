import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Entity from '../Entity';

it('renders correctly', () => {
    entityData = {"id":2,"initials":"FEUP"};
    const tree = renderer.create(<Entity data={entityData}/>).toJSON();

    expect(tree).toMatchSnapshot();
});