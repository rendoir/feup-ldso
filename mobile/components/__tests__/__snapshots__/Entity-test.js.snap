// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 0.5,
          "borderColor": "#c9c9c9",
          "flexDirection": "row",
          "marginLeft": 16,
          "paddingBottom": 13,
          "paddingRight": 16,
          "paddingTop": 13,
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
              "paddingLeft": 5,
            },
          ]
        }
        uppercase={false}
      >
        FEUP
      </Text>
    </View>
  </View>
</View>
`;
